rule_test.go

```diff
@@ -1038,6 +1038,42 @@ func TestRE(t *testing.T) {
 	}
 }
 
+func TestLastContent(t *testing.T) {
+	for _, tt := range []struct {
+		rule string
+		want *Content
+	}{
+		{
+			rule: `alert udp $HOME_NET any -> $EXTERNAL_NET any (sid:1337; msg:"foo"; content:"bar";)`,
+			want: &Content{Pattern: []byte("bar")},
+		},
+		{
+			rule: `alert udp $HOME_NET any -> $EXTERNAL_NET any (sid:1337; msg:"foo"; content:"bar"; pcre:"/foo.*bar/iU"; content:"foo"; within:40; pcre:"/foo.*bar.*baz/iU";)`,
+			want: &Content{
+				Pattern: []byte("foo"),
+				Options: []*ContentOption{
+					{
+						Name:  "within",
+						Value: "40",
+					},
+				},
+			},
+		},
+		{
+			rule: `alert udp $HOME_NET any -> $EXTERNAL_NET any (sid:1337; msg:"foo";)`,
+			want: nil,
+		},
+	} {
+		r, err := ParseRule(tt.rule)
+		if err != nil {
+			t.Fatalf("re: parse rule failed: %v", err)
+		}
+		if got := r.LastContent(); !reflect.DeepEqual(got, tt.want) {
```

Comment:
Looks great, last nit the other test files moved from reflect.DeepEqual to using "github.com/kylelemons/godebug/pretty" as it makes diffing in tests much more readable.
Can you do the same here? You can find examples in https://github.com/google/gonids/blob/master/optimize_test.go

Reply:
This package seems nice indeed :-)